### PR TITLE
fix(rollup): fixup dist and bundle in Buffer

### PR DESCRIPTION
**Pull request by edaniels**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,4 +13,7 @@ npm-debug.log
 builderror.log
 
 bson.sublime-project
-bson.sublime-workspace
\ No newline at end of file
+bson.sublime-workspace
+
+dist/*
```
Comment (daprahamian) [inline .gitignore]: Do we want to permanently remove the `dist` directory and only have the files in the published version?

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ npm run build
 A simple example of how to use BSON in the browser:
 
 ```html
-<script src="./dist/bson.js"></script>
+<script src="./dist/bson.bundle.js"></script>
```
Comment (daprahamian) [inline README.md]: Can the `README.md` include more documentation about what each `dist` file is?

File: lib/bson/bson.js [hunk 3]
```diff
@@ -383,6 +384,7 @@ module.exports.Symbol = Symbol;
 module.exports.BSON = BSON;
 module.exports.DBRef = DBRef;
 module.exports.Binary = Binary;
+module.exports.ObjectId = ObjectID;
```
Comment (daprahamian) [inline lib/bson/bson.js]: This seems like an unrelated change.

File: lib/bson/parser/deserializer.js [hunk 4]
```diff
@@ -27,17 +28,18 @@ var deserialize = function(buffer, options, isArray) {
     throw new Error(`bson size must be >= 5, is ${size}`);
   }
 
-  if (options.allowObjectSmallerThanBufferSize && buffer.length < size) {
-    throw new Error(`buffer length ${buffer.length} must be >= bson size ${size}`);
+  if (options.allowObjectSmallerThanBufferSize && Buffer.byteLength(buffer) < size) {
```
Comment (daprahamian) [inline lib/bson/parser/deserializer.js]: Why `Buffer.byteLength`?

File: lib/bson/parser/deserializer.js [hunk 5]
```diff
@@ -27,17 +28,18 @@ var deserialize = function(buffer, options, isArray) {
...
+    throw new Error(`buffer length ${Buffer.byteLength(buffer)} must be >= bson size ${size}`);
   }
 
-  if (!options.allowObjectSmallerThanBufferSize && buffer.length !== size) {
-    throw new Error(`buffer length ${buffer.length} must === bson size ${size}`);
+  if (!options.allowObjectSmallerThanBufferSize && Buffer.byteLength(buffer) !== size) {
+    console.log(buffer, buffer.length, Buffer.byteLength(buffer));
```
Comment (daprahamian) [inline lib/bson/parser/deserializer.js]: `console.log`

File: .gitignore (hunk 1 above)
Comment (edaniels) [reply .gitignore]: I think the bundle is the one good file to keep here

File: README.md (hunk 2 above)
Comment (edaniels) [reply README.md]: I don't think it's necessary since only 1 is committed. It should be self documenting from package.json.

File: lib/bson/bson.js (hunk 3 above)
Comment (edaniels) [reply lib/bson/bson.js]: It's related in removing confusing duplication in index.js. This is the one diffed thing not present.

File: lib/bson/parser/deserializer.js (hunk 4 above)
Comment (edaniels) [reply lib/bson/parser/deserializer.js]: `length` is not a real property on `Buffer`. It comes from the underlying Uint8Array which is not always consistent depending on the environment. `byteLength` will always produce the correct result which will be the same as `length`.

File: lib/bson/parser/deserializer.js (hunk 5 above)
Comment (edaniels) [reply lib/bson/parser/deserializer.js]: whoops

**Comment by edaniels** [comment]
Changes addressed
@daprahamian AFAICT, Buffer does not get globally assigned if you look at the output of each dist. It appears to be used locally. Even if it did, a good citizen polyfill would be the same API and functionality and would only polyfill in the event it's already present.

**Comment by edaniels** [comment]
I've also loaded both bson and mongodb-extjson in the same page with 0 errors.

File: README.md (hunk 2 above)
Comment (daprahamian) [reply README.md]: Since I don't fully understand it, I think it would be helpful to document what each bundle is in the readme.

Comment (edaniels) [reply README.md]: Done.
